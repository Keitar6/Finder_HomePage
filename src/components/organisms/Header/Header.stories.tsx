import type { Meta, StoryObj } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';
import Header from './Header';

const meta = {
  title: 'Organisms/Header',
  component: Header,
  tags: ['autodocs'],
  decorators: [(Story) => <BrowserRouter>{Story()}</BrowserRouter>],
} satisfies Meta<typeof Header>;

export default meta;
type Story = StoryObj<typeof meta>;

export const defaultHeader: Story = {};
