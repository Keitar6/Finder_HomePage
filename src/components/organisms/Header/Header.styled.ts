import styled from 'styled-components';
import { PaletteType } from '../../../theme/theme.types';

export const HeaderWrapper = styled.div<{ palette: PaletteType }>`
  background-color: ${({ palette }) => palette.light};
  background: linear-gradient(
    180deg,
    ${({ palette }) => palette.light} 0%,
    ${({ palette }) => palette.blueLight} 100%
  );
  margin-bottom: 50px;
`;
export const HeaderNavBar = styled.nav`
  align-items: center;
  display: flex;
  justify-content: center;
  width: 100%;
  padding: 8px 16px;
`;

export const HeaderBox = styled.div`
  align-items: center;
  display: flex;
  justify-content: center;
  width: 100%;
  padding: 5rem 0px;
`;

export const Logo = styled.img`
  height: 64px;
`;
export const LogoAndHambuergerContainer = styled.div`
  display: flex;
  width: 100%;
  align-items: center;
  gap: 1rem;
`;
export const SearchAndButtonsContainer = styled.div`
  display: flex;
  width: 100%;
  justify-content: center;
  gap: 1rem;
  align-items: center;
`;
export const ButtonsWrapper = styled.div`
  display: flex;
  gap: 0.5rem;
`;
export const Banner = styled.div`
  display: flex;
  padding: 0 20px;
  justify-content: center;
  align-items: center;
  max-width: 50rem;
`;

export const DownloadWrapper = styled.div`
  display: flex;
  position: relative;
  justify-content: center;
  align-items: center;
  gap: 0.5rem;
`;
