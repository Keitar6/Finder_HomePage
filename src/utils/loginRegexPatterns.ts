import { InputVariant, RegexPat } from '../components/atoms/Input/Input.types';
import { Getters } from '../types/shared.types';

type LoginRegexProps = Getters<InputVariant, RegexPat>;

const loginRegexPattern: LoginRegexProps = {
  search: {
    pattern: /^[a-zA-Z0-9]+$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
};

export default loginRegexPattern;
