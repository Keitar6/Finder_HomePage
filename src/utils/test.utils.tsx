import { PropsWithChildren } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import ThemeProvider from '../theme/ThemeContext';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  initialUrl?: string;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    initialUrl = '/',
    ...renderOptions
  }: ExtendedRenderOptions = {},
) {
  const Wrapper = ({ children }: PropsWithChildren<object>): JSX.Element => (
    <ThemeProvider>
        {children}
    </ThemeProvider>
  );

  return { ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from '@testing-library/react';
