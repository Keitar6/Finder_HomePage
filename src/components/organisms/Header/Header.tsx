import Finder from '../../../assets/img/Finder.svg';
import { useContext, useState } from 'react';
import { ThemeContext } from '../../../theme/ThemeContext';
import Typography from '../../atoms/Typography/Typography';
import * as Styled from './Header.styled';
import Icon from '../../atoms/Icon/Icon';
import Button from '../../atoms/Button/Button';
import SearchBar from '../../molecules/SearchBar/SearchBar';
import QRDropdownItem from '../../molecules/QRDropdownItem/QRDropdownItem';

const Header = () => {
  const { palette } = useContext(ThemeContext);
  const [isDropdownOpen, setIsDropdownOpen] = useState<boolean>(false);

  const openDropdown = () => {
    setIsDropdownOpen(true);
  };
  const closeDropdown = () => {
    setIsDropdownOpen(false);
  };

  return (
    <Styled.HeaderWrapper palette={palette}>
      <Styled.HeaderNavBar>
        <Styled.LogoAndHambuergerContainer>
          <div>Hamburger Icon</div>
          <Styled.Logo src={Finder} alt='Logo Finder' />
        </Styled.LogoAndHambuergerContainer>
        <Styled.SearchAndButtonsContainer>
          <Styled.DownloadWrapper onMouseOver={openDropdown} onMouseOut={closeDropdown}>
            <Icon name='download' />
            <Typography color='blueDark' variant='boldParagraph'>
              App
            </Typography>
            {isDropdownOpen ? <QRDropdownItem /> : null}
          </Styled.DownloadWrapper>

          <SearchBar placeholder='Search' />
          <div>Placeholder Icon Bell </div>

          <Styled.ButtonsWrapper>
            <Button variant='ghost' style={{ fontWeight: 600 }}>
              Log in
            </Button>
            <Button variant='primary' style={{ fontWeight: 600 }}>
              Join
            </Button>
          </Styled.ButtonsWrapper>
        </Styled.SearchAndButtonsContainer>
      </Styled.HeaderNavBar>

      <Styled.HeaderBox>
        <Styled.Banner>
          <Typography style={{ textAlign: 'center' }} variant='banner'>
            Start making better money decisions right now
          </Typography>
        </Styled.Banner>
      </Styled.HeaderBox>
    </Styled.HeaderWrapper>
  );
};

export default Header;
