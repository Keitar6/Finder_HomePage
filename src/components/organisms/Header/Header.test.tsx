import { cleanup } from '@testing-library/react';
import { renderWithProviders } from '../../../utils/test.utils';
import Header from './Header';

describe('Header', () => {
  afterEach(cleanup);
  it('renders correctly', () => {
    const snapshot = renderWithProviders(<Header />, {});

    expect(snapshot).toMatchSnapshot();
  });
});
