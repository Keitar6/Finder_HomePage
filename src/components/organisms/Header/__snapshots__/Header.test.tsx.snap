// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header headers renders correctly for logged out user 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-aXZVg lhKbmH"
      >
        <nav
          class="sc-gEvEer eiCYYm"
        >
          <div
            class="sc-dcJsrY yrNFI"
          >
            <div
              class="sc-tagGq kwBrw"
            >
              <span />
            </div>
            <img
              alt="Finders logo"
              class="sc-fqkvVR fVIsRy"
              src="Finder.svg"
            />
          </div>
          <div
            class="sc-iGgWBj cpqCjJ"
          >
            <div
              class="sc-bmzYkS fkBqwy"
            >
              <span />
              <p
                style="font-family: Roboto, Arial, sans-serif; font-weight: 500; font-size: 1rem; line-height: 1rem; color: rgb(2, 113, 225);"
              >
                App
              </p>
            </div>
            <div>
              <div
                class="sc-cwHptR RUXrT"
              >
                <div
                  class="sc-jEACwC bjXzFf"
                >
                  <span />
                  <input
                    class="sc-cPiKLX TSGZv"
                    data-testid="SearchBarInput"
                    placeholder="Search"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="sc-feUZmu doXeFE"
            >
              <div
                class="sc-fHjqPf kwXyHW"
              >
                <button
                  aria-label="more than 99 notifications"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiBadge-root css-ui7n8a-MuiBadge-root"
                  >
                    <span />
                    <span
                      class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError css-to4x1n-MuiBadge-badge"
                    >
                      2
                    </span>
                  </span>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <div
              class="sc-gsFSXq dFhXeg"
            >
              <button
                class="sc-kpDqfm sc-dAlyuH AHXat gvBhsO"
                style="font-weight: 600;"
              >
                Log in
              </button>
              <button
                class="sc-kpDqfm kBKlPr"
                style="font-weight: 600;"
              >
                Join
              </button>
            </div>
          </div>
        </nav>
        <div
          class="sc-eqUAAy iNvBfU"
        >
          <div
            class="sc-kAyceB dxBABw"
          >
            <h1
              style="font-family: Roboto, Arial, sans-serif; font-weight: 700; font-size: 53px; line-height: 55px; color: rgb(36, 42, 47); text-align: center;"
            >
              Start making better money decisions right now
            </h1>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-aXZVg lhKbmH"
    >
      <nav
        class="sc-gEvEer eiCYYm"
      >
        <div
          class="sc-dcJsrY yrNFI"
        >
          <div
            class="sc-tagGq kwBrw"
          >
            <span />
          </div>
          <img
            alt="Finders logo"
            class="sc-fqkvVR fVIsRy"
            src="Finder.svg"
          />
        </div>
        <div
          class="sc-iGgWBj cpqCjJ"
        >
          <div
            class="sc-bmzYkS fkBqwy"
          >
            <span />
            <p
              style="font-family: Roboto, Arial, sans-serif; font-weight: 500; font-size: 1rem; line-height: 1rem; color: rgb(2, 113, 225);"
            >
              App
            </p>
          </div>
          <div>
            <div
              class="sc-cwHptR RUXrT"
            >
              <div
                class="sc-jEACwC bjXzFf"
              >
                <span />
                <input
                  class="sc-cPiKLX TSGZv"
                  data-testid="SearchBarInput"
                  placeholder="Search"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="sc-feUZmu doXeFE"
          >
            <div
              class="sc-fHjqPf kwXyHW"
            >
              <button
                aria-label="more than 99 notifications"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiBadge-root css-ui7n8a-MuiBadge-root"
                >
                  <span />
                  <span
                    class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError css-to4x1n-MuiBadge-badge"
                  >
                    2
                  </span>
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="sc-gsFSXq dFhXeg"
          >
            <button
              class="sc-kpDqfm sc-dAlyuH AHXat gvBhsO"
              style="font-weight: 600;"
            >
              Log in
            </button>
            <button
              class="sc-kpDqfm kBKlPr"
              style="font-weight: 600;"
            >
              Join
            </button>
          </div>
        </div>
      </nav>
      <div
        class="sc-eqUAAy iNvBfU"
      >
        <div
          class="sc-kAyceB dxBABw"
        >
          <h1
            style="font-family: Roboto, Arial, sans-serif; font-weight: 700; font-size: 53px; line-height: 55px; color: rgb(36, 42, 47); text-align: center;"
          >
            Start making better money decisions right now
          </h1>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-aXZVg fRtWVR"
      >
        <nav
          class="sc-gEvEer eiCYYm"
        >
          <div
            class="sc-dcJsrY yrNFI"
          >
            <div
              class="sc-eBMEME hmSOMb"
            >
              <span />
            </div>
            <img
              alt="Finders logo"
              class="sc-fqkvVR fVIsRy"
              src="Finder.svg"
            />
          </div>
          <div
            class="sc-iGgWBj cpqCjJ"
          >
            <div
              class="sc-bmzYkS fkBqwy"
            >
              <span />
              <p
                style="font-family: Roboto, Arial, sans-serif; font-weight: 500; font-size: 1rem; line-height: 1rem; color: rgb(2, 113, 225);"
              >
                App
              </p>
            </div>
            <div>
              <div
                class="sc-cwHptR RUXrT"
              >
                <div
                  class="sc-jEACwC bjXzFf"
                >
                  <span />
                  <input
                    class="sc-cPiKLX TSGZv"
                    data-testid="SearchBarInput"
                    placeholder="Search"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="sc-feUZmu doXeFE"
            >
              <div
                class="sc-fHjqPf kwXyHW"
              >
                <button
                  aria-label="more than 99 notifications"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiBadge-root css-ui7n8a-MuiBadge-root"
                  >
                    <span />
                    <span
                      class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError css-to4x1n-MuiBadge-badge"
                    >
                      2
                    </span>
                  </span>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <div
              class="sc-gsFSXq dFhXeg"
            >
              <button
                class="sc-kpDqfm sc-dAlyuH AHXat gvBhsO"
                style="font-weight: 600;"
              >
                Log in
              </button>
              <button
                class="sc-kpDqfm kBKlPr"
                style="font-weight: 600;"
              >
                Join
              </button>
            </div>
          </div>
        </nav>
        <div
          class="sc-eqUAAy iNvBfU"
        >
          <div
            class="sc-kAyceB dxBABw"
          >
            <h1
              style="font-family: Roboto, Arial, sans-serif; font-weight: 700; font-size: 53px; line-height: 55px; color: rgb(36, 42, 47); text-align: center;"
            >
              Start making better money decisions right now
            </h1>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-aXZVg fRtWVR"
    >
      <nav
        class="sc-gEvEer eiCYYm"
      >
        <div
          class="sc-dcJsrY yrNFI"
        >
          <div
            class="sc-eBMEME hmSOMb"
          >
            <span />
          </div>
          <img
            alt="Finders logo"
            class="sc-fqkvVR fVIsRy"
            src="Finder.svg"
          />
        </div>
        <div
          class="sc-iGgWBj cpqCjJ"
        >
          <div
            class="sc-bmzYkS fkBqwy"
          >
            <span />
            <p
              style="font-family: Roboto, Arial, sans-serif; font-weight: 500; font-size: 1rem; line-height: 1rem; color: rgb(2, 113, 225);"
            >
              App
            </p>
          </div>
          <div>
            <div
              class="sc-cwHptR RUXrT"
            >
              <div
                class="sc-jEACwC bjXzFf"
              >
                <span />
                <input
                  class="sc-cPiKLX TSGZv"
                  data-testid="SearchBarInput"
                  placeholder="Search"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="sc-feUZmu doXeFE"
          >
            <div
              class="sc-fHjqPf kwXyHW"
            >
              <button
                aria-label="more than 99 notifications"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiBadge-root css-ui7n8a-MuiBadge-root"
                >
                  <span />
                  <span
                    class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError css-to4x1n-MuiBadge-badge"
                  >
                    2
                  </span>
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="sc-gsFSXq dFhXeg"
          >
            <button
              class="sc-kpDqfm sc-dAlyuH AHXat gvBhsO"
              style="font-weight: 600;"
            >
              Log in
            </button>
            <button
              class="sc-kpDqfm kBKlPr"
              style="font-weight: 600;"
            >
              Join
            </button>
          </div>
        </div>
      </nav>
      <div
        class="sc-eqUAAy iNvBfU"
      >
        <div
          class="sc-kAyceB dxBABw"
        >
          <h1
            style="font-family: Roboto, Arial, sans-serif; font-weight: 700; font-size: 53px; line-height: 55px; color: rgb(36, 42, 47); text-align: center;"
          >
            Start making better money decisions right now
          </h1>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
